src/web/components/KeyPairs/KeyPairEditDialog.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import { SubmitHandler } from 'react-hook-form';
+
+import { EditKeyPairFormDTO } from '../../../api/services/adminServiceHelpers';
+import { Dialog } from '../Core/Dialog';
+import { Form } from '../Core/Form';
+import { TextInput } from '../Input/TextInput';
+import { KeyPairModel } from './KeyPairModel';
+
+import '../ApiKeyManagement/KeyEditDialog.scss';
+
+export type OnKeyPairEdit = (
+  form: EditKeyPairFormDTO,
+  updateKeyPair: React.Dispatch<React.SetStateAction<KeyPairModel>>
+) => void;
+
+type KeyPairEditDialogProps = Readonly<{
+  onEdit: OnKeyPairEdit;
+  triggerButton: JSX.Element;
+  keyPair: KeyPairModel;
+}>;
+
+function KeyPairEditDialog({
+  onEdit,
+  triggerButton,
+  keyPair: keyPairInitial,
+}: KeyPairEditDialogProps) {
+  const [open, setOpen] = useState(false);
+  const [keyPair, updateKeyPair] = useState<KeyPairModel>(keyPairInitial);
```

Comment:
i think it's standard to call the state setter function set###().  So setKeyPair() here